python/cudf/cudf/core/column/string.py

```diff
@@ -4930,20 +4945,18 @@ def to_array(self, fillna: bool = None) -> np.ndarray:
 
         return self.to_arrow().to_pandas().values
 
-    def __array__(self, dtype=None):
-        raise TypeError(
-            "Implicit conversion to a host NumPy array via __array__ is not "
-            "allowed, Conversion to GPU array in strings is not yet "
-            "supported.\nTo explicitly construct a host array, "
-            "consider using .to_array()"
-        )
```

Comment:
These and the other implicit conversion methods shouldn't be removed unless they're already implemented in the Column base class?

Reply:
I don't think we need these in `Column` at all. `Frame` and its subclasses are the user-facing classes, whereas `Column` is meant to be internal. `Frame` does give the appropriate warning, so I don't think we need this unless we anticipate an API where users can access columns.

Reply:
It's moreso preventing us as developers from accidentally iterating over Columns in host code or accidentally trying to convert a Column to a numpy array for example. Without these we can fall into very bad codepaths of calling `__getitem__` with a scalar over and over.

Reply:
Got it, I left `__cuda_array_interface__` implemented as raising `NotImplementedError` in the base class precisely because I thought developers might be doing `cupy.asarray` and we wanted to discourage that. Currently there is no protection against `__array__` except for string columns, so I assumed that this simply wasn't an issue. I'll add this into the base class for safety.